test/common/router/retry_state_impl_test.cc

```diff
@@ -1313,7 +1368,7 @@ TEST_F(RouterRetryStateImplTest, HostSelectionAttempts) {
   EXPECT_EQ(2, state_->hostSelectionMaxAttempts());
 }
 
-TEST_F(RouterRetryStateImplTest, Cancel) {
+TEST_F(RouterRetryStateImplTest, CancelDuringExponentialRetryBackoff) {
```

Comment:
I've significantly expanded on the retry cancellation tests since I feel that's a bit of a hole in the current test suite and also where the most risk is for this PR. I want to make sure there's no situation where resources or stats are miscounted.